Devscord.DiscordFramework/Integration/Services/DiscordClientRolesService.cs

```diff
@@ -57,12 +58,20 @@ public async Task SetRolePermissions(IEnumerable<ChannelContext> channels, Disco
         {
             await Task.Delay(1000);
             var createdRole = this.GetSocketRoles(server.Id).FirstOrDefault(x => x.Id == role.Id);
-            var channelPermissions = new OverwritePermissions(permissions.AllowPermissions.GetRawValue(), permissions.DenyPermissions.GetRawValue());
+            if (createdRole == null)
+            {
+                Log.Error("Created role was null");
+                return;
+            }
 
+            var channelPermissions = new OverwritePermissions(permissions.AllowPermissions.GetRawValue(), permissions.DenyPermissions.GetRawValue());
             Parallel.ForEach(channels, async c =>
             {
                 var channelSocket = (IGuildChannel) await this._discordClientChannelsService.GetChannel(c.Id);
-                await channelSocket.AddPermissionOverwriteAsync(createdRole, channelPermissions);
+                if (channelSocket.PermissionOverwrites.All(x => x.TargetId != createdRole.Id))
+                {
+                    await channelSocket.AddPermissionOverwriteAsync(createdRole, channelPermissions);
```

Comment:
z tego co słyszałem to w parallel lepiej używać .Result / Wait()
ale możesz doczytać o tym 

bo ten wątek i tak jest w tle
a obsługa każdego dodatkowego wątku zabiera moc

jak kod jest źle napisany to 50% albo 80% mocy może pójść na obsługe faktu że w działa to wielowątkowo 
ale tylko tak słyszałem, dobrze byłoby to potwierdzić